docs/api/api/api.txt

```diff
@@ -1847,6 +1847,76 @@ PUT /message/?project=<project>&package=<package>
 XmlBody :message
 XmlResult: message
 
+== Status Checks
+  WARNING: This API is currently in beta and unstable
+
+=== Repositories
+
+<project>: The name of a project, e.g. home:user1
+<repository>: The name of a repository of the project, e.g. openSUSE_Tumbleweed
+<uuid>: The build id you have received for the repository, e.g. 12345
+
+GET /status_reports/published/<project>/<repository>/reports/<uuid>
+  Get a status report with this uuid
+XmlResult: status_report
+
+
+PUT /status_reports/published/<project>/<repository>/reports/<uuid>
+  Create or update check data
+XmlResult: check
+
+=== Requests
+
+<request>: The number of the request, e.g. 12345
+
+GET /status_reports/requests/<request>/reports
+  Get a the status report for this request
+XmlResult: status_report
+
+
+PUT /status_reports/requests/<request>/reports
+  Create or update check data
+XmlResult: check
+
+== Required Status Checks
+  WARNING: This API is currently in beta and unstable
+
+=== Projects
+
+<project>: The name of a project, e.g. home:user1
+<check>: The name of the required check
+
+GET /status_reports/projects/<project>/required_checks
+  Get all required checks for this project
+XmlResult: required_checks
+
+POST /status_reports/projects/<project>/required_checks/<check>
+  Create a required check
+XmlResult: required_checks
+
+DELETE /status_reports/projects/<project>/required_checks/<check>
+  Delete a required check
+XmlResult: required_checks
+
+=== Repositories
+
+<project>: The name of a project, e.g. home:user1
+<repository>: The name of a repository of the project, e.g. openSUSE_Tumbleweed
+<check>: The name of the required check
+
+GET /status_reports/repositories/<project>/<repository>/required_checks
+  Get all required checks for this repository
+XmlResult: required_checks
+
+
+POST /status_reports/repositories/<project>/<repository>/required_checks/<check>
+  Create a required check
+XmlResult: required_checks
+
+
+DELETE /status_reports/repositories/<project>/<repository>/required_checks/<check>
+  Delete a required check
+XmlResult: required_checks
```

Comment:
The required checks belong to the project configuration, so they should be stored to the backend (`/source/<project>/_project/...` most likely). Means that the (published?) repository binding would need to be part of the xml. However, it would have the advantages
* One place for all required check definitions (with published/build/source/... places in mind)
* Usual handling with all kind of operations (log/diff/blame/project-copy/...) without any extra effort

Reply:
Please let's discuss functional changes somewhere else. I'm trying to document what is there. Not how it should be. I will open an issue.